rosidl_runtime_py/rosidl_runtime_py/convert.py

```diff
@@ -85,41 +92,67 @@ def to_string(val):
                         val += '...'
             r = str(val)
         else:
-            r = message_to_csv(val, truncate_length)
+            r = message_to_csv(val, truncate_length, no_arr, no_str)
         return r
     result = ''
     # We rely on __slots__ retaining the order of the fields in the .msg file.
     for field_name in msg.__slots__:
         value = getattr(msg, field_name)
+        field_type = msg._fields_and_field_types[field_name[1:]]
```

Comment:
@vinnamkim consider querying for the field type only if and when it's actually necessary

Reply:
In the updated commit, `field_type` is passed to `to_string()` and if-statement to check `no_arr` is moved to `to_string()`. It becomes inevitable to query for the field type only if and when it's actually necessary.